### PR TITLE
Add test for metastore access count in Delta Lake

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add test for metastore access count in Delta Lake

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Comment by findepi** [comment]
@ebyhr see CI for delta-lake failured

**Comment by ebyhr** [comment]
@findepi Fixed CI failure.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/CountingAccessFileHiveMetastore.java [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+
+@ThreadSafe
+public class CountingAccessFileHiveMetastore
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/CountingAccessFileHiveMetastore.java:45]: This is copied from Iceberg's `CountingAccessFileHiveMetastore`, right?

Let's move the class to trino-hive/src/test/java and share via a test-jar dependency.
When moving, drop "File" from the name (it's a left over)

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/InternalDeltaLakeConnectorFactory.java [hunk 2]
```diff
@@ -71,7 +72,8 @@ public static Connector createConnector(
             String catalogName,
             Map<String, String> config,
             ConnectorContext context,
-            Module module)
+            Module module,
+            Optional<HiveMetastore> metastore)
```
Comment (findepi) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/InternalDeltaLakeConnectorFactory.java:76]: add before `Module module,` (as you did in `TestingDeltaLakePlugin` class)

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakePerTransactionMetastoreCache.java [hunk 3]
```diff
@@ -98,6 +98,7 @@ private DistributedQueryRunner createQueryRunner(boolean enablePerTransactionHiv
         DistributedQueryRunner queryRunner = DistributedQueryRunner.builder(session).build();
 
         queryRunner.installPlugin(new TestingDeltaLakePlugin(
+                Optional.empty(),
```
Comment (findepi) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakePerTransactionMetastoreCache.java:101]: as a followup, this test class can perhaps be simplified with the `CountingAccess{File}HiveMetastore` class.
i.e. we could remove reflection use from here.

for that, we would need to ensure every method in `CountingAccess{File}HiveMetastore` is counted, not only selected ones.

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/InternalDeltaLakeConnectorFactory.java (hunk 2 above)
Comment (findepi) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/InternalDeltaLakeConnectorFactory.java:76]: actually, i think the change here is not required.
You can bind your own metastore via `Module module`, you just need to set `hive.metastore` to some unsupported value (eg `other`).

would that work? i think this is what `TestDeltaLakeMetastoreAccessOperations` is doing.

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/CountingAccessFileHiveMetastore.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/CountingAccessFileHiveMetastore.java]: Right, moved to Hive module.

**Review by findepi — APPROVED** [review]

**Pull request merged**